packages/perseus/src/widgets/radio/base-radio.jsx

```diff
@@ -243,13 +243,25 @@ function BaseRadio(props: Props): React.Node {
     const shouldShowInstructions = !sat;
 
     const responsiveClassName = css(styles.responsiveFieldset);
+    // Note(TB): Each set of questions needs its own unique name to be
+    // considered a radio group; this is what allows only one choice at
+    // a time to be selected for single select questions. Tried
+    // useUniqueIdWithMock from WB, but encountered errors we could not fix.
+    // When fixed, can replace generateQuestionId with the WB function.
+    let id = 0;
+    const generateQuestionId = () => {
+        return `questionId-${Date.now()}-${id++}`;
+    };
+    const questionId = generateQuestionId();
```

Comment:
Declaring these in the body of the component won't have the effect you're expecting. Think about the render flow:

1. `BaseComponent` is called
2. `id` is defined as 0
3. `generateQuestionId` is declared
4. `generateQuestionId` is called which sets `questionId` to `"questionId-1668630158-1"`
5. `BaseComponent` is called again
6. `id` is defined as 0
7. `generateQuestionId` is declared
8. `generateQuestionId` is called which sets `questionId` to `"questionId-1668630180-1"`

You'll need to pull lines 251-254 out of the body.

---

Also I know I recommended this code, but it's important to note that this is kind of a gross hack. If I didn't recommend it, I'd probably make a case against it 😂 One reason is that it's possible for `id` to grow really big and reach a max number...in practice this would be unlikely. We could safeguard against this with the modulo operator, but I think that might be overkill since  you already marked this as a temporary hack. (NAN on this paragraph, just wanted to take blame.)

Reply:
Thank you so much for pointing this out! It allowed me to super simplify the IDs. It is temporary until useUniqueId works again, but it's less complicated than using the date thing.